src/utils/SpokeUtils.ts

```diff
@@ -102,3 +103,66 @@ export function isZeroAddress(address: string): boolean {
 export function getMessageHash(message: string): string {
   return isMessageEmpty(message) ? ZERO_BYTES : keccak256(message as Hex);
 }
+
+export async function findInvalidFills(spokePoolClients: {
+  [chainId: number]: SpokePoolClient;
+}): Promise<InvalidFill[]> {
+  const invalidFills: InvalidFill[] = [];
+
+  // Iterate through each spoke pool client
+  for (const spokePoolClient of Object.values(spokePoolClients)) {
+    // Get all fills for this client
+    const fills = spokePoolClient.getFills();
+
+    // Process each fill
+    for (const fill of fills) {
+      // Skip fills with unsafe deposit IDs
+      if (isUnsafeDepositId(fill.depositId)) {
+        continue;
+      }
+
+      // Get all deposits (including duplicates) for this fill's depositId, both in memory and on-chain
+      const depositResult = await spokePoolClients[fill.originChainId].findAllDeposits(fill.depositId);
+
+      // If no deposits found at all
+      if (!depositResult.found) {
```

Comment:
nb. RE mis-specified chain IDs, there might be something to consider here:

```suggestion
      const depositResult = await spokePoolClients[fill.originChainId]?.findAllDeposits(fill.depositId);

      // If no deposits found at all
      if (!depositResult?.found) {
```

Reply:
[f9d939e](https://github.com/across-protocol/sdk/pull/1085/commits/f9d939e0c89510228a95daa7e1bf92fcba695e5f)